From 6543660c429224f3cb8c9d16c786a521d0fc2969 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:06 +0100
Subject: [PATCH] tracking-2.6.25-changed-s3c2410_dma_request-reurn.patch

s3c2410_dma_request used to return 0 for OK and something else
for error, now it returns -ve error code or +ve dma channel index + flag

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/sdio/hcd/s3c24xx/s3c24xx_hcd.c |    3 ++-
 sound/soc/s3c24xx/s3c24xx-pcm.c        |    2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/sdio/hcd/s3c24xx/s3c24xx_hcd.c b/drivers/sdio/hcd/s3c24xx/s3c24xx_hcd.c
index 8c3bbbc..af0066d 100644
--- a/drivers/sdio/hcd/s3c24xx/s3c24xx_hcd.c
+++ b/drivers/sdio/hcd/s3c24xx/s3c24xx_hcd.c
@@ -1192,7 +1192,8 @@ static int s3c24xx_hcd_hw_init(struct s3c24xx_hcd_context * context)
 
 	}
 
-	if (s3c2410_dma_request(context->dma_channel, &s3c24xx_hcd_dma_client, NULL)) {
+	if (s3c2410_dma_request(context->dma_channel, &s3c24xx_hcd_dma_client,
+								    NULL) < 0) {
 		DBG_PRINT(SDDBG_ERROR, ("unable to get DMA channel.\n"));
 		status = -ENOENT;
 		goto out_free_dma;
diff --git a/sound/soc/s3c24xx/s3c24xx-pcm.c b/sound/soc/s3c24xx/s3c24xx-pcm.c
index 7806ae6..40abef1 100644
--- a/sound/soc/s3c24xx/s3c24xx-pcm.c
+++ b/sound/soc/s3c24xx/s3c24xx-pcm.c
@@ -172,7 +172,7 @@ static int s3c24xx_pcm_hw_params(struct snd_pcm_substream *substream,
 					  prtd->params->client, NULL);
 
 		if (ret < 0) {
-			DBG(KERN_ERR "failed to get dma channel\n");
+			DBG(KERN_ERR "failed to get dma channel: %d\n", ret);
 			return ret;
 		}
 	}
-- 
1.5.6.3

